src/etl/SystemState.hpp

```diff
@@ -33,9 +37,19 @@ struct SystemState {
      * In strict read-only mode, the process will never attempt to become the ETL writer, and will only publish ledgers
      * as they are written to the database.
      */
-    bool isReadOnly = false;
+    util::prometheus::Bool isReadOnly = PrometheusService::boolMetric(
+        "etl_read_only",
```

Comment:
better change it to read_only to distinguish the etl which is not writing at the moment